src/librustdoc/core.rs

```diff
@@ -129,6 +129,7 @@ pub fn run_core(search_paths: SearchPaths, cfgs: Vec<String>, externs: Externs,
     let sess = session::build_session_(sessopts, cpath, diagnostic_handler,
                                        codemap, cstore.clone());
     rustc_lint::register_builtins(&mut sess.lint_store.borrow_mut(), Some(&sess));
+    write::init_llvm(&sess);
```

Comment:
I would prefer it a lot (if you remember, it was my original approach), but AFAICT it requires moving the `init_llvm` code in order to avoid cyclic dependencies between `librustc` (where `Session` is constructed) and `librustc_trans`. Is that acceptable? Otherwise, is there a clever way to workaround the cyclic dependency without moving the code? (actually I would like it if it was not too clever... I would hate it to be unintuitive)